File: python/1726.py
from collections import defaultdict
from typing import List

class Solution:   
    def tupleSameProduct(self, nums: List[int]) -> int:
        m = defaultdict(int)
        for i in range(len(nums)):
            for j in range(i+1, len(nums)):
                m[nums[i] * nums[j]] += 1
        return sum(x * (x - 1) // 2 for x in m.values()) << 3

s = Solution()
nums = [1,2,4,5,10]
print(s.tupleSameProduct(nums))

File: go/1726.go
package main

func tupleSameProduct(nums []int) int {
	// var m map[int]int
	m := map[int]int{}
	for i := 0; i < len(nums); i++ {
		for j := i + 1; j < len(nums); j++ {
			m[nums[i]*nums[j]]++
		}
	}
	ans := 0
	for _, v := range m {
		ans += v * (v - 1) / 2
	}
	return ans << 3
}
